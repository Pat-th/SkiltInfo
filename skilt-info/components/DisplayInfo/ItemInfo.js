import React from 'react';
import { StyleSheet, Text, View, Clipboard, ToastAndroid, Vibration } from 'react-native';
import { TouchableHighlight, TouchableOpacity } from 'react-native-gesture-handler';
import Colors from "../../Constants/Colors";

const ItemInfo = props => {

    const longPressHandler = async () => {
        console.log("Long press!");
        await Clipboard.setString(props.id + " " + props.value);
        ToastAndroid.show("Kopiert til utklippstavle", ToastAndroid.SHORT);
        Vibration.vibrate(30);
    }

    return(
            <View style={styles.card}>
                <TouchableOpacity onLongPress={() => longPressHandler()}>
                    <View>
                        <Text style={styles.title}>{props.id}</Text>
                        <Text style={styles.value}>{props.value}</Text>
                    </View>
                </TouchableOpacity>
            </View>
    );
};

const styles = StyleSheet.create({
    card: {
        //Shadow props only work on IOS
        shadowColor: "black",
        shadowOffset: { width: 0, height: 2},
        shadowRadius: 6,
        shadowOpacity: 0.26,
        elevation: 5, //Shadow on Android
        backgroundColor: "white",
        padding: 10,
        borderRadius: 10, //Rounded corners
        marginTop: 10,
        marginLeft: 10,
        marginRight: 10,
        backgroundColor: Colors.primaryColor1
    },
    title: {
        fontSize: 18
    }
});

export default ItemInfo;