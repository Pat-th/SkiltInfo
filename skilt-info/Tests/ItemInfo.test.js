import React from "react";
import renderer from "react-test-renderer";

import ItemInfo from "../components/DisplayInfo/ItemInfo";

describe("<ItemInfo />", () => {
    it("Has 2 children", () => {
        const tree = renderer.create(<ItemInfo />).toJSON();
        expect(tree.children.length).toBe(2);
    });
    it('Renders correctly', () => {
        const tree = renderer.create(<ItemInfo />).toJSON();
        expect(tree).toMatchSnapshot();
      });
   });