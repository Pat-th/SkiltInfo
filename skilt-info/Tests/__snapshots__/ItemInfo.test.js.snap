// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemInfo /> Renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "borderColor": "black",
        "marginLeft": 10,
        "marginRight": 10,
        "padding": 10,
      }
    }
  >
    <View />
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "fontSize": 18,
              }
            }
          />
          <Text />
        </View>
      </View>
    </RNGestureHandlerButton>
  </View>
  <View
    style={
      Object {
        "borderBottomColor": "#a38e8c",
        "borderBottomWidth": 0.8,
        "marginLeft": 18,
        "marginRight": 18,
      }
    }
  />
</View>
`;
